python/whylogs/extras/image_metric.py

```diff
@@ -101,32 +102,65 @@ def image_based_metadata(img):
     }
 
 
+@dataclass(frozen=True)
+class ImageMetricConfig(MetricConfig):
+    allowed_exif_tags: Set[str] = field(default_factory=set)
+    forbidden_exif_tags: Set[str] = field(default_factory=set)
+
+
 class ImageMetric(CompoundMetric):
-    def __init__(self, submetrics: Dict[str, Metric]):
+    def __init__(
+        self,
+        submetrics: Dict[str, Metric],
+        allowed_exif_tags: Optional[Set[str]] = None,
+        forbidden_exif_tags: Optional[Set[str]] = None,
+    ):
         if ImageType is None:
             logger.error("Install Pillow for image support")
         super(ImageMetric, self).__init__(submetrics)
+        self._allowed_exif_tags = allowed_exif_tags or set()
+        self._forbidden_exif_tags = forbidden_exif_tags or set()
 
     @property
     def namespace(self) -> str:
         return "image"
 
+    def _wants_to_track(self, exif_tag: str) -> bool:
+        if exif_tag in self._forbidden_exif_tags:
+            return False
+        if not self._allowed_exif_tags:  # empty set -> allow anything not explicitly forbidden
+            return True
+        return exif_tag in self._allowed_exif_tags
+
     def columnar_update(self, view: PreprocessedColumn) -> OperationResult:
         submetric_names = self.submetrics.keys()
         count = 0
         for image in list(chain.from_iterable(view.raw_iterator())):
             if isinstance(image, ImageType):
                 metadata = get_pil_image_metadata(image)
                 for attr in metadata.keys():
+                    if attr not in submetric_names:  # EXIF tag discovery
```

Comment:
This is the key new behavior we need: discover and track features when found in images.